AoE2ScenarioParser/datasets/support/info_dataset_base.py

```diff
@@ -1,73 +1,254 @@
+from __future__ import annotations
 from enum import Enum
-from typing import List
+from typing import Union
 
 
 class InfoDatasetBase(Enum):
+
+    """
+
+    This enum class is the base class for the UnitInfo, BuildingInfo, HeroInfo and OtherInfo datasets. It provides the
+    properties and functions common to all the mentioned classes
+
+    **Methods**
+    >>> InfoDatasetBase.from_id()
+    >>> InfoDatasetBase.from_dead_id()
+    >>> InfoDatasetBase.from_icon_id()
+    >>> InfoDatasetBase.from_hotkey_id()
+    >>> InfoDatasetBase.gaia_only()
+    >>> InfoDatasetBase.non_gaia()
+
+    """
+
     @staticmethod
-    def _id_map():
+    def _id_map() -> dict:
+
+        """
+
+        PRIVATE METHOD!
```

Comment:
Also don't think this has to be added in the doc string, espeically not with full caps 11. The underscore makes it clear & in any api doc it will be marked as private too.

Reply:
Fixing in the new commit 11

Reply:
Fixing these in the new commit 11